### PR TITLE
Docs refactor

### PR DESCRIPTION
**Pull request by ferd**

### CODE REVIEW

File: content/en/docs/dependencies.md [hunk 1]
```diff
@@ -110,7 +73,75 @@ This will allow the flexibility to write and generate software where various dis
...
+
+{{< blocks/callout type="info" title="Treating Conflicts as Errors">}}
+ If you ever want rebar3 to abort as soon as it detects a dependency conflict, instead of skipping the file and proceeding as usual, add the line <code>{deps_error_on_conflict, true}.</code> to your rebar configuration file.
+{{< /blocks/callout >}}
+
+For convenience reasons (and because [hex.pm](https://hex.pm) mandates semver) hex dependencies can be specified using semver-like syntax:
+
+```erlang
+{deps,[
+  rebar,                   % fetches latest known version, ignoring pre-releases
```
Comment (tsloughter) [inline content/en/docs/dependencies.md:108]: Something I was starting to change else where is the use of `rebar` as a hex package in examples. I think better would be to use an actual package like `elli` with real version examples.

File: content/en/docs/dependencies.md [hunk 2]
```diff
@@ -110,7 +73,75 @@ This will allow the flexibility to write and generate software where various dis
...
+
+```
+_checkouts
+└── depA
+    └── src
+```
+
+Any application or plugin in `_checkouts` will take precedence over the same application if it is additionally listed in the `rebar.config`'s `deps`, `plugins`, or `project_plugins`. This also overrides anything already fetched to `_build`.
+
+Note that `_checkout` is an override, this means that for it to work a `dep` or `plugin` entry in rebar.config _must_ exist.
```
Comment (tsloughter) [inline content/en/docs/dependencies.md:142]:
```suggestion
Note that `_checkouts` is an override, this means that for it to work a `dep` or `plugin` entry in rebar.config _must_ exist.
```

File: content/en/docs/dependencies.md (hunk 1 above)
Comment (ferd) [reply content/en/docs/dependencies.md:108]: Fair. I wouldn't block on it for this, but we should probably find something like that to work from.

I think when we wrote the initial docs we just didn't have a lot of packages with many versions to go around but we could use one now.

Comment (tsloughter) [reply content/en/docs/dependencies.md:108]: Yea, agreed. I can take care of it when I do a full sweep of replacing `rebar` as the dep example.

**Pull request merged**